lib/src/solver/solve_suggestions.dart

```diff
@@ -0,0 +1,214 @@
+        return null;
+      }
+      if (stripLowerBound) {
+        updatedPackageVersions.sort((a, b) => a.name.compareTo(b.name));
+        final formattedConstraints = updatedPackageVersions
+            .map((e) => packageAddDescription(entrypoint, e))
+            .join(' ');
+        return _ResolutionSuggestion(
+            '* Try updating the following constraints: $topLevelProgram pub add $formattedConstraints',
+            priority: 4);
+      } else {
+        return _ResolutionSuggestion(
+            '* Try an upgrade of your constraints: $topLevelProgram pub upgrade --major-versions',
+            priority: 4);
+      }
+    }
+    return null;
+  }
+
+  final visited = <String>{};
+  final stopwatch = Stopwatch()..start();
+  final suggestions = <_ResolutionSuggestion>[];
+  void addSuggestion(_ResolutionSuggestion? suggestion) {
+    if (suggestion != null) suggestions.add(suggestion);
+  }
+
+  for (final externalIncompatibility
+      in incompatibility.externalIncompatibilities) {
+    if (stopwatch.elapsed > Duration(seconds: 3)) {
+      // Never spend more than 3 seconds computing suggestions.
+      break;
+    }
+    final cause = externalIncompatibility.cause;
+    if (cause is SdkCause) {
+      addSuggestion(await suggestSdkUpdate(cause));
+    } else {
+      for (final term in externalIncompatibility.terms) {
+        final name = term.package.name;
+
+        if (!visited.add(name)) {
+          continue;
+        }
+        addSuggestion(await suggestSinglePackageUpdate(name));
+      }
+    }
+  }
+  if (suggestions.isEmpty) {
+    addSuggestion(await suggestUnlockingAll(stripLowerBound: true) ??
+        await suggestUnlockingAll(stripLowerBound: false));
+  }
+
+  if (suggestions.isEmpty) return null;
+  final tryOne = suggestions.length == 1
+      ? 'You can try the following suggestion to make the pubspec resolve:'
+      : 'You can try one of the following suggestions to make the pubspec resolve:';
+
+  suggestions.sort((a, b) => a.priority.compareTo(b.priority));
+
+  return '\n$tryOne\n${suggestions.take(5).map((e) => e.suggestion).join('\n')}';
+}
+
+class _ResolutionSuggestion {
+  final String suggestion;
+  final int priority;
+  _ResolutionSuggestion(this.suggestion, {this.priority = 0});
+}
+
+Future<SolveResult?> tryResolve(SolveType type, SystemCache cache, Package root,
+    {LockFile? lockFile,
+    Iterable<String> unlock = const [],
+    Map<String, Version> sdkOverrides = const {}}) async {
+  try {
+    return await resolveVersions(type, cache, root,
+        lockFile: lockFile, sdkOverrides: sdkOverrides);
+  } on SolveFailure {
+    return null;
+  }
+}
+
+String packageAddDescription(Entrypoint entrypoint, PackageId id) {
+  final name = id.name;
+  final isDev = entrypoint.root.pubspec.devDependencies.containsKey(name);
+  final constraint = VersionConstraint.compatibleWith(id.version);
+  final devPart = isDev ? 'dev:' : '';
+  return '$devPart$name:$constraint';
```

Comment:
This won't work if it's not a hosted dependency from the default server.
Should we make some check and avoid proposing suggestions for other than that kind of packages?

Reply:
I instead added code, so we can update packages from any server

Also I now skip non-hosted dependencies, as they will never resolve anyway.